_posts/2021/2021-11-15-dont-use-floats-use-ruby-money-for-representing-currency.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Don't Use Floats and Use the Ruby Money Gem to Represent Currencies"
+description: "It's surprising how often you still see this in the wild"
+layout: article
+category: ruby
+image:
+  base: "2021/dont-use-floats-use-ruby-money-for-representing-currency"
+  alt: "Coins"
+  credit: "Dan Dennis"
+  source: "https://unsplash.com/photos/pZ56pVKd_6c"
+
+---
+
+We often have to manage monetary values in our applications. And there’s a terrific open source solution in the Ruby community that is _much_ better than rolling your own.
+
+
+## Instead of…
+
+…useing `Float`s or `BigDecimal`s to represent money
+
+```ruby
+# In your migration
+add_column :products, :price, :decimal
+
+class Product < ApplicationModel
+  def price_to_s
+    "$ #{price.round(2).to_s("F")} USD"
+  end
+end
+
+product.price = 5
+#=> 0.5e1
+product.price_to_s
+#=> "$ 5.0 USD"
+```
+
+## Use…
+
+…the [ruby `money`](https://github.com/RubyMoney/money) gem, or in Rails the [`money-rails` gem](https://github.com/RubyMoney/money-rails):
+
+```ruby
+# In your migration
+add_monetize :products, :price
+
+class Product < ApplicationModel
+  monetize :price_cents
+end
+
+product.price = Money.from_amount(5, "USD")
+#=> #<Money fractional:500 currency:USD>
+product.price.format
+#=> "$5.00"
+```
+
+
+## Why?
+
+The first part of this recommendation is to use a non-floating point implementation to represent your your currency values. Floats can introduce rounding errors, due to their [underlying representaion inside a computer](http://download.oracle.com/docs/cd/E19957-01/806-3568/ncg_goldberg.html). Just _do not_ do this.
```

Comment:
Is there a smoother, clearer way to start this paragraph?

*double 'your'

*no comma after 'errors'